lib/revocation.js

```diff
@@ -0,0 +1,135 @@
+var assert = require('assert')
+    , async = require('async')
+    , aws = require('aws-sdk')
+    , LRU = require('lru-cache')
+    ;
+
+var dynamo;
+var lru; // LRU cache of token revocation information
+
+exports.init = function (options) {
+    assert.ok(options);
+    assert.ok(options.config);
+    assert.ok(options.logger);
+
+    if (!lru) {
+        lru = LRU({
+            max: +options.config.max_revocation_cache_size.current,
+            maxAge: 1000 * +options.config.max_revocation_cache_age.current
+        });
+        options.config.max_revocation_cache_size.on('modified', function (new_max) {
+            lru.max = +new_max;
+            options.logger.info({ max: +new_max }, 'setting new maximum size of the revocation check LRU cache');
+        });
+        options.config.max_revocation_cache_age.on('modified', function (new_age) {
+            lru._maxAge = 1000 * +new_age;
+            options.logger.info({ max_age: +new_age }, 'setting new maximum age in seconds of the revocation check LRU cache');
+        });
+    }
+
+    if (!dynamo) {
+        // Create DynamoDB client 
+        aws.config.accessKeyId = options.config.aws_access_key_id.current;
+        aws.config.secretAccessKey = options.config.aws_secret_access_key.current;
+        aws.config.region = options.config.aws_revocation_region.current;
+        aws.config.sslEnabled = true;
+        dynamo = new aws.DynamoDB({ params: { TableName: options.config.aws_revocation_table.current }});
+    }
+
+    return {
+
+        check_revocation: function (req_id, jwt, callback) {
+            // Check revocation status of the JWT token and its issuance chain
+            var ca = [];
+            if (jwt.payload.ca)
+                jwt.payload.ca.forEach(function (jti) { ca.push(jti); });
+            if (jwt.payload.jti)
+                ca.push(jwt.payload.jti);
+            var valid = true;
```

Comment:
A token without jti would have been rejected prior to this revocation check since we don't issue such tokens, and we only accept tokens issued by us. But at the theoretical level I can think of a jti-less token as un-revocable, so it would mean it passes the check.